source/opt/ssa_rewrite_pass.cpp

```diff
@@ -615,7 +628,17 @@ Pass::Status SSARewriter::RewriteFunctionIntoSSA(Function* fp) {
             << fp->PrettyPrint(0) << "\n";
 #endif
 
-  if (modified) pass_->context()->KillDebugDeclareInsts(fp);
+  if (modified) {
+    // Kill DebugDeclare only if it is added as DebugValue for each store
+    // or phi instruction and there is no store or phi instruction that it
+    // cannot be added as DebugValue because the out-of-scope issue.
```

Comment:
Could you clarify this comment?  I don't understand what it's trying to state.

Reply:
Fundamentally, I don't think I'm understanding why `DebugValue` would have to remove a `DebugDeclare`.  I'm missing something, but I'm not sure what.